### PR TITLE
Switch to the Semver4j fork

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Bower.kt [hunk 1]
```diff
@@ -21,8 +21,6 @@ package org.ossreviewtoolkit.analyzer.managers
 
 import com.fasterxml.jackson.databind.JsonNode
 
-import com.vdurmont.semver4j.Requirement
```
Comment (MarcelBochtler) [inline analyzer/src/main/kotlin/managers/Bower.kt:24]: Commit message: You're never referencing `[1]`.

File: analyzer/src/main/kotlin/managers/Npm.kt [hunk 2]
```diff
@@ -145,7 +146,7 @@ open class Npm(
 
     override fun command(workingDir: File?) = if (Os.isWindows) "npm.cmd" else "npm"
 
-    override fun getVersionRequirement(): Requirement = Requirement.buildNPM("6.* - 8.*")
+    override fun getVersionRequirement(): RangesList = RangesListFactory.create("6.* - 8.*")
```
Comment (MarcelBochtler) [inline analyzer/src/main/kotlin/managers/Npm.kt:150]: Note for myself: The NPM ranges are covered by the [XRangeProcessor](https://github.com/semver4j/semver4j/blob/2258ea7b3aa3a43afa8ce41203af9ac844b95d4f/src/main/java/org/semver4j/internal/range/processor/XRangeProcessor.java).

File: cli/src/main/kotlin/commands/RequirementsCommand.kt [hunk 3]
```diff
@@ -120,7 +122,7 @@ class RequirementsCommand : OrtCommand(
                             val actualVersion = tool.getVersion()
                             runCatching {
                                 val isRequiredVersion = tool.getVersionRequirement().let {
-                                    it == CommandLineTool.ANY_VERSION || it.isSatisfiedBy(actualVersion)
+                                    Semver.coerce(actualVersion).satisfies(it)
```
Comment (MarcelBochtler) [inline cli/src/main/kotlin/commands/RequirementsCommand.kt:125]: Why is the check for `ANY_VERSION` dropped?

File: downloader/src/main/kotlin/VersionControlSystem.kt [hunk 4]
```diff
@@ -404,7 +404,7 @@ abstract class VersionControlSystem {
      * Check whether the VCS tool is at least of the specified [expectedVersion], e.g. to check for features.
      */
     fun isAtLeastVersion(expectedVersion: String): Boolean {
-        val actualVersion = Semver(getVersion(), Semver.SemverType.LOOSE)
-        return actualVersion.isGreaterThanOrEqualTo(expectedVersion)
+        val actualVersion = Semver.coerce(getVersion())
+        return actualVersion.isGreaterThanOrEqualTo(Semver.coerce(expectedVersion))
```
Comment (MarcelBochtler) [inline downloader/src/main/kotlin/VersionControlSystem.kt:408]: Note for myself: `SemverType.LOOSE` was implicitly used before as well for `expectedVersion`.

File: model/src/main/kotlin/PackageCuration.kt [hunk 5]
```diff
@@ -65,10 +64,7 @@ data class PackageCuration(
      */
     private fun isApplicableIvyVersion(pkgId: Identifier) =
         runCatching {
-            // TODO: This check does not completely comply to the Ivy version-matchers specification. E.g. the version
-            //       '1.0' does not satisfy the version range [1.0,2.0], see
-            //       https://github.com/vdurmont/semver4j/issues/67.
-            Requirement.buildIvy(id.version).isSatisfiedBy(Semver(pkgId.version, Semver.SemverType.LOOSE))
+            Semver.coerce(pkgId.version).satisfies(id.version)
```
Comment (MarcelBochtler) [inline model/src/main/kotlin/PackageCuration.kt:67]: `satisfies(String)` never uses `coerce` internally.
Isn't this required to solve the `TODO` above?

File: cli/src/main/kotlin/commands/RequirementsCommand.kt (hunk 3 above)
Comment (sschuberth) [reply cli/src/main/kotlin/commands/RequirementsCommand.kt:125]: `ANY_VERSION` was introduced here in d8c65c3f8ac329fc89c01f2694b03f5cf6d41309 to work around a limitation of the original `isSatisfiedBy()` call to not handle non-semantic versions, but even those should be accepted here. This is not necessary anymore with the new code as it uses `coerce()` before.

File: model/src/main/kotlin/PackageCuration.kt (hunk 5 above)
Comment (sschuberth) [reply model/src/main/kotlin/PackageCuration.kt:67]: It shouldn't be necessary (anymore), see https://github.com/semver4j/semver4j/issues/100. And I just manually verified that `Semver.coerce("1.0").satisfies("[1.0,2.0]") == true`.

**Review by MarcelBochtler — APPROVED** [review]

**Pull request merged**